src/puppetlabs/trapperkeeper/services_internal.clj

```diff
@@ -29,23 +28,67 @@
        (symbol? (first sig))
        (vector? (second sig))))
 
-(defn fns-map?
-  "A predicate to determine whether or not an object is a map of fns, as
-  used internally by the `service` macro"
-  [m]
-  (and (map? m)
-       (every? keyword? (keys m))
-       (every? seq? (vals m))
-       (every? fn-sig? (apply concat (vals m)))))
+(def Protocol
+  "A schema to determine whether or not an object is a protocol definition."
+  (schema/pred protocol?))
 
-(defn var->symbol
+(def Symbol
```

Comment:
Alternatively, `clojure.lang.Symbol` can just be used directly as a schema instead of defining this.